### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by dmikusa-pivotal**
Initial set up of the buildpack. Includes:

- a Makefile for easy building & linting (including tools setup)
- a lint config, taken from libcnb
- a build script, which is compatible with pipeline-builder
- README & license files
- vscode settings, ensures imports order consistently
- a skeleton profile module with basic libcnb detect/build funcs
- test setup for profile module using go's built-in testing & gomega matchers

Signed-off-by: Daniel Mikusa <dmikusa@vmware.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module github.com/buildpacks/profile
+
+go 1.17
```
Comment (gcemaj) [inline go.mod:3]: any reason to not use 1.18?

Comment (dmikusa-pivotal) [reply go.mod:3]: I don't have a strong preference. On the Paketo Java buildpacks we're on 1.17, so I just used that out of habit. Our rationale is that we don't want to run the absolute latest version of Go, we don't often need those features and don't want any potential bugs/issues, so we hang back a release.

Comment (gcemaj) [reply go.mod:3]: 👍 , makes sense. I dont have the power to approve and merge but this looks like a sensible template to build upon.

**Comment by dmikusa-pivotal** [comment]
@samj1912 Does this look OK to you as a starting place? Two of the three failing jobs are due to the labels. I believe when this merges, the CI job should run to update the list of labels. Until that happens, you can't pick the right labels to satisfy the jobs. 

The third is missing password information. It needs Docker Creds. I'm not sure what to use there. If there are creds as Secrets, I can switch to use those. If that's something we need to work out still, that's fine too. Just let me know either way. Thanks

File: .github/pipeline-descriptor.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+codeowners:
+- path:  "*"
+  owner: "@buildpacks/buildpack-authors-tooling-maintainers"
+
+package:
+  repository:     docker.io/buildpacks/profile
+  register:       true
+  registry_token: ${{ secrets.GITHUB_TOKEN }}
+
+docker_credentials:
```
Comment (samj1912) [inline .github/pipeline-descriptor.yml:14]: Could we just publish to ghcr?

Comment (dmikusa-pivotal) [reply .github/pipeline-descriptor.yml:14]: Follow up from the BAT team meeting. We discussed and I believe the decision was to go ahead with using Docker Hub. We have other resources published there, so it makes sense to publish buildpack images there as well.

File: .github/pipeline-descriptor.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+github:
+  username: ${{ secrets.GITHUB_USERNAME }}
+  token:    ${{ secrets.GITHUB_TOKEN }}
+
+codeowners:
+- path:  "*"
+  owner: "@buildpacks/buildpack-authors-tooling-maintainers"
+
+package:
+  repository:     docker.io/buildpacks/profile
```
Comment (samj1912) [inline .github/pipeline-descriptor.yml:10]:
```suggestion
  repository:     docker.io/buildpacksio/profile
```

File: .github/workflows/create-package.yml [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                    done
+
+                else
+                  pack buildpack package \
+                    "${PACKAGE}:${VERSION}" \
+                    --config "${HOME}"/package.toml \
+                    --format "${FORMAT}"
+                fi
+              env:
+                PACKAGES: docker.io/buildpacks/profile
```
Comment (samj1912) [inline .github/workflows/create-package.yml:191]:
```suggestion
                PACKAGES: docker.io/buildpacksio/profile
```

Comment (samj1912) [reply .github/workflows/create-package.yml:191]: @dmikusa-pivotal I have added the appropriate secrets to the repository. Can you please make sure that the dockerhub org is buildpacksio and not buildpacks? Thanks.

Comment (dmikusa-pivotal) [reply .github/workflows/create-package.yml]: OK, creds look good. I fixed a few and have double-checked that all of the images are `docker.io/buildpacksio` and code references `github.com/buildpacks`.

**Comment by dmikusa-pivotal** [comment]
I faked out the label jobs by adding in the expected labels manually. When this is merged, it should update the full set of labels available to the repo & this won't be an issue going forward.

I think we're probably good to merge this. @samj1912, @gcemaj or @ekcasey - any last-minute thoughts?

**Review by samj1912 — APPROVED** [review]

**Pull request merged**